### PR TITLE
lpm11/develop

### PR DESCRIPTION
**Pull request by lpm11**
- タイマー (120秒)
- 開始まで5秒空ける
- 徐々にお題が明らかになる
- 回答は `「」` なし、文字列完全一致で判定
- 回答時の獲得点数は残り時間、描き手にも加点
- サウンド
  - 出題時
  - 正解時
  - 時間切れ時
- メッセージを本家っぽく

### CODE REVIEW

File: client/index.html [hunk 1]
```diff
@@ -1,9 +1,17 @@
 <html>
   <head>
+    <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width,initial-scale=1.0,maximum-scale=1.0,user-scalable=no">
     <script src="/index.js"></script>
     <link rel="stylesheet" type="text/css" href="/index.css">
   </head>
+  <div id="subject-wrapper">
```
Comment (tomykaira) [inline client/index.html]: These wrappers should be included in `main`

File: client/index.js [hunk 2]
```diff
@@ -195,6 +195,38 @@ function body(name) {
...
   };
+
+  const subjectElement = document.querySelector('#subject');
+  eventHandlerMap['setSubject'] = (data) => {
+    subjectElement.innerText = data.subject;
+  };
+
+  const timeElement = document.querySelector('#time');
+  var countDownTimer = null;
+  var countDownEndTime = null;
```
Comment (tomykaira) [inline client/index.js]: Please use `let` instead of `var`.

File: client/index.js [hunk 3]
```diff
@@ -195,6 +195,38 @@ function body(name) {
...
+  eventHandlerMap['startTimer'] = (data) => {
+    if (countDownTimer !== null) {
+      clearInterval(countDownTimer);
+      countDownTimer = null;
+    }
+
+    countDownEndTime = (new Date()).getTime() + data.timeLimit;
```
Comment (tomykaira) [inline client/index.js]: nitpick: () around `new Date()` is not necessary.
`Date.now()` does the same.

Comment (tomykaira) [inline client/index.js]: This variable is not used out of this scope.
`var` declaration in l.206 is unnecessary.

File: server/index.js [hunk 4]
```diff
@@ -2431,19 +2440,26 @@ app.ws('/', (ws, req) => {
           break;
         }
       }
-      if (/^「.*」$/.test(decoded.msg) && answerer === players[painterIndex])  {
-        sendOne(ws, error('絵師は「」つき発言をできません。'));
-        return;
+      if (answerer === players[painterIndex]) {
+        decoded.msg = decoded.msg.replace(currentWord, '○○○');
```
Comment (tomykaira) [inline server/index.js]: When you repeat the word twice, the latter is not masked.
e.g. `なぷきんなぷきん→○○○なぷきん`

File: server/index.js [hunk 5]
```diff
@@ -2431,19 +2440,26 @@ app.ws('/', (ws, req) => {
...
-        sendScoreBoard(answerer.name + 'さんが正解しました!1ポイント獲得! ');
-        changePainter((painterIndex + 1) % players.length);
+        let point = Math.round((gameEndTime - Date.now()) / 1000);
+        stopGameTimer();
+        stopRevealer();
+
+        fanOut(endTimer());
+        fanOut(setSubject(currentWord));
+        fanOut(systemMessage(answerer.name + ' が正解！「' + currentWord + '」+ ' + point + '点'));
+        fanOut(systemMessage('絵を描いた人 ' + answerer.name + ' + ' + point + '点'));
```
Comment (tomykaira) [inline server/index.js]: `answerer -> painter`?

**Review by tomykaira — COMMENTED** [review]
Basically OK.
There will be lots of corner cases, but we do not touch them now.

File: client/index.html (hunk 1 above)
Comment (lpm11) [reply client/index.html]: Fixed, moved into `#main`. Also wrappers are renamed to `container`s.

File: client/index.js (hunk 2 above)
Comment (lpm11) [reply client/index.js]: Fixed.

File: client/index.js (hunk 3 above)
Comment (lpm11) [reply client/index.js]: Fixed :P

Comment (lpm11) [reply client/index.js]: Fixed.

File: server/index.js (hunk 4 above)
Comment (lpm11) [reply server/index.js]: [Oh sorry](http://dic.nicovideo.jp/a/pat).

File: server/index.js (hunk 5 above)
Comment (lpm11) [reply server/index.js]: Fixed :P

**Comment by tomykaira** [comment]
Thx. great.

**Pull request merged**